src/child.h

```diff
@@ -2,42 +2,41 @@
 #define __HLWM_CHILD_H_
 
 #include "object.h"
+#include <memory>
 
 template<typename T>
 class Child_ {
-    // implement a pointer to a child object
-    // in the object tree.
-    // if this is assigned a new value, then the child object
-    // in the owner is updated automatically
+    // implement a static child object in the object tree. Static means that
+    // init() is called on start up and reset() is called on shutdown.
```

Comment:
If you're changing the comment anyway, you could turn it into a proper Doxygen docstring by putting it in front of the class declaration in this format:
```
/*!
 * ...
 */